src/components/IconButton.tsx

```diff
@@ -0,0 +1,57 @@
+import { cva, type VariantProps } from "class-variance-authority";
+import { forwardRef } from "react";
+import { cn } from "~/lib/utils";
+import type { IconType } from "react-icons";
+import { useRouter } from "next/router";
+
+const iconButtonVariants = cva("font-orbitron flex items-center", {
+  variants: {
+    variant: {
+      primary:
+        "text-red-500 [text-shadow:_0_0_10px_#FF0000,0_0_20px_#FF0000,0_0_30px_#FF0000] filter drop-shadow-[0_0_10px_#FF0000]",
+    },
+  },
+  defaultVariants: {
+    variant: "primary",
+  },
+});
+
+export interface IconButtonProps
+  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
+    VariantProps<typeof iconButtonVariants> {
+  className?: string;
+  icon: IconType;
```

Comment:
If possible, allow this `icon` to be passed as a string property as well instead of the icon directly
change to:
```
icon: IconType | string;
```
then in the component parse string into IconType (if icon is of type string)

Reply:
Not required anymore, skipping